### ISSUE TITLE
fields API returns value for a sub-field with ignore_above set

### ISSUE THREAD

**Issue by astefan** [elastic/elasticsearch#61028]
Given the mapping:

```
{
	"mappings": {
          "properties": {
            "text_field": {
             "type": "text",
             "fields": {
               "keyword_subfield": {
                 "type": "keyword",
                 "ignore_above": 10
               }
             }
           }
	 }
	}
}
```

and test data like:

```
{"index":{}}
{"text_field" : "1234567890 asdlkfjasdf"}
{"index":{}}
{"text_field" : "foo"}
```

For a query of the form:

```
{
    "size": 1000,
    "_source": false,
    "fields": [
        {
            "field": "text_field"
        },
        {
            "field": "text_field.keyword_subfield"
        }
    ]
}
```

The output shows a value for the `keyword_subfield`  which, according to `ignore_above: 10` should have no value:

```
        "hits": [
            {
                "_index": "test",
                "_id": "Zq3O4nMBFdprR7a5qr4z",
                "_score": 1,
                "fields": {
                    "text_field.keyword_subfield": [
                        "1234567890 asdlkfjasdf"
                    ],
                    "text_field": [
                        "1234567890 asdlkfjasdf"
                    ]
                }
            },
            {
                "_index": "test",
                "_id": "Z63O4nMBFdprR7a5qr40",
                "_score": 1,
                "fields": {
                    "text_field.keyword_subfield": [
                        "foo"
                    ],
                    "text_field": [
                        "foo"
                    ]
                }
            }
        ]
```

**Comment by astefan** [comment]
@jtibshirani

**Comment by jtibshirani** [comment]
This is the same underlying problem as #61033, I'll close and track the fix in #61033.
